File: Samples/BasicApplication/BasicApplication/BasicApplication.Web/Modules/Administration/Language/LanguageForm.cs

namespace BasicApplication.Administration.Forms
{
    using Serenity;
    using Serenity.ComponentModel;
    using Serenity.Data;
    using System;
    using System.ComponentModel;
    using System.Collections.Generic;
    using System.IO;

    [FormScript("Administration.Language")]
    [BasedOnRow(typeof(Entities.LanguageRow))]
    public class LanguageForm
    {
        public String LanguageId { get; set; }
        public String LanguageName { get; set; }
    }
}

File: serene/src/Serene.Web/Modules/ServerTypes/Administration/LanguageForm.ts
import { initFormType, PrefixedContext, StringEditor } from "@serenity-is/corelib";

export interface LanguageForm {
    LanguageId: StringEditor;
    LanguageName: StringEditor;
}

export class LanguageForm extends PrefixedContext {
    static readonly formKey = 'Administration.Language';
    declare private static init: boolean;

    constructor(...args: ConstructorParameters<typeof PrefixedContext>) {
        super(...args);

        if (!LanguageForm.init) {
            LanguageForm.init = true;

            initFormType(LanguageForm, [
                'LanguageId', StringEditor,
                'LanguageName', StringEditor
            ]);
        }
    }
}